apps/mobile-wallet/src/features/ecosystem/authorizedConnections/AuthorizedConnectionsScreen.tsx

```diff
@@ -0,0 +1,113 @@
+import dayjs from 'dayjs'
+import { useTranslation } from 'react-i18next'
+import { Image } from 'react-native'
+import styled from 'styled-components/native'
+
+import AddressBadge from '~/components/AddressBadge'
+import AppText from '~/components/AppText'
+import Button from '~/components/buttons/Button'
+import FlashListScreen from '~/components/layout/FlashListScreen'
+import ListItem from '~/components/ListItem'
+import { connectionRemoved } from '~/features/ecosystem/authorizedConnections/authorizedConnectionsActions'
+import { selectAllAuthorizedConnections } from '~/features/ecosystem/authorizedConnections/authorizedConnectionsSelectors'
+import { AuthorizedConnection } from '~/features/ecosystem/authorizedConnections/authorizedConnectionsTypes'
+import { useAppDispatch, useAppSelector } from '~/hooks/redux'
+import { DEFAULT_MARGIN, VERTICAL_GAP } from '~/style/globalStyle'
+import { ImpactStyle, vibrate } from '~/utils/haptics'
+import { showToast } from '~/utils/layout'
+
+const AuthorizedConnectionsScreen = () => {
+  const dispatch = useAppDispatch()
+  const { t } = useTranslation()
+  const connections = useAppSelector(selectAllAuthorizedConnections)
+
+  const handleDisconnectPress = (host: string) => {
+    vibrate(ImpactStyle.Light)
+    dispatch(connectionRemoved(host))
+    showToast({
+      text1: t('Revoked connection to {{ host }}', { host }),
+      type: 'success'
+    })
+  }
+
+  return (
+    <FlashListScreen
+      data={connections}
+      estimatedItemSize={70}
+      headerOptions={{ headerTitle: t('Preauthorized connections'), type: 'stack' }}
+      screenTitle={t('Preauthorized connections')}
+      screenIntro={t('Manage dApp connections')}
+      contentPaddingTop
+      renderItem={({ item: connection, index }) => (
+        <ListItemStyled
+          key={connection.host}
```

Comment:
What if we connect to the same host on both mainnet and testnet, or with different addresses? Will we have clashing react keys then?